esy-lib/win32_remove_readonly.c

```diff
@@ -0,0 +1,84 @@
+#include <stdio.h>
+
+#include <caml/alloc.h>
+#include <caml/memory.h>
+#include <caml/mlvalues.h>
+
+#ifdef WIN32
+#include <Windows.h>
+#define limit 32767
+void directory_get_entries_recursively(const char* path) {
+
+  DWORD attrs = GetFileAttributesA(path);
+  if (attrs == INVALID_FILE_ATTRIBUTES) {  
+    printf("GetFileAttributesA(%s) (in main) failed\n", path); 
+    return;
+  }
+
+  if (! (attrs & FILE_ATTRIBUTE_DIRECTORY)) {
+    if (attrs & FILE_ATTRIBUTE_READONLY) {
+      /* printf("Readonly %s\n", path); */
+      /* printf("Attrs: %x \nRemoving readonly attribute\n", attrs); */
+      SetFileAttributesA(path, attrs & (~FILE_ATTRIBUTE_READONLY));
+    } else {
+      /* printf("RW %s\n", path); */
+    }
+    /* DeleteFileA(path); */
+    return;
+  }
+  
+  WIN32_FIND_DATA ffd;
+  char search_path[limit];
+  strcpy(search_path, path);
+  strcat(search_path, "\\*");
+  HANDLE hFind = FindFirstFile(search_path, &ffd);
+  if (INVALID_HANDLE_VALUE == hFind) {
+    /* printf("FindFirstFile failed for %s\n", path); */
+    return;
+  }
+  do {
```

Comment:
I wonder if this could be made simpler, in general, by recursing using the Libuv file operations (and could live in the OCaml side); https://github.com/aantron/luv/blob/dfab9dfbb6802d6663347b5f25af7fcf19a68914/src/file.mli#L1009

The `chmod` API supports clearing the read-only bit on windows.